### ISSUE TITLE
Convert to .NET Core

### ISSUE THREAD

**Issue by micdenny** [EasyNetQ/EasyNetQ.Management.Client#34]
connected to https://github.com/EasyNetQ/EasyNetQ/issues/508 and the PR https://github.com/EasyNetQ/EasyNetQ/pull/606

**Comment by conniey** [comment]
@micdenny 
To help with testability of the .NET Core migration, I ended up moving the test assets to NUnit 3, so you can run `dotnet test`.  I also added AppVeyor integration, so the test results are pushed back to the service.  
*  [Associated code changes](https://github.com/conniey/EasyNetQ.Management.Client/compare/feature/netcore...conniey:migrateToNUnit3)
*  [Example AppVeyor build](https://ci.appveyor.com/project/conniey/easynetq-management-client/build/1.0.18/tests)

Are you interested in this accepting this PR?  If so, where should I merge my changes into?

Cheers,
Connie

**Comment by conniey** [comment]
Also, what tasks are left before this can be accepted into master? I ask, because I am tasked with moving EasyNetQ to .NET Core and saw that this was one of the tasks in [#508](https://github.com/EasyNetQ/EasyNetQ/issues/508)

**Comment by micdenny** [comment]
> Are you interested in this accepting this PR? If so, where should I merge my changes into?

Yes, thank you very much for this PR, I absolutely interested in this, just be patient because this week I don't have to much time for ENQ, I can't guarantee I can look at this week, feel free to ping me back next week if I forgot to look at it.

> Also, what tasks are left before this can be accepted into master?

Yes, that's the list of TODO for ENQ, where one of that was porting EasyNetQ.Management.Client, then there's some code to fix on ENQ net core like the connection string management, check the TODO in [RabbitHutch](https://github.com/EasyNetQ/EasyNetQ/pull/606/files#diff-e178600c720410c7b2a2fc7d27dc1996).

**Comment by zidad** [comment]
@micdenny it looks good to me, shall we just merge this into feature/netcore as it isn't stable yet anyway? looks like a great improvement!

**Comment by micdenny** [comment]
> shall we just merge this into feature/netcore as it isn't stable yet anyway?

@zidad ok fair enough 😄 let's merge in

**Comment by conniey** [comment]
Is there a NuGet feed that we can test the .NET Core version of EasyNetQ.Management.Client on?  

My teammate and I are moving the main library to .NET Core and the tests have a dependency on the Management Client.

**Comment by micdenny** [comment]
@conniey I can manually push to MyGet, does appveyor built a nuget package for the latest commit?

**Comment by conniey** [comment]
Yeah it does. From the latest PR #40, the [artifacts here](https://ci.appveyor.com/project/EasyNetQ/easynetq-management-client/build/1.0.112-wjjaflbx/artifacts); underneath the Artifacts tab.

I was also able to add a CI (on my test feed) with these additional lines: [appveyor.yml L25-31](https://github.com/conniey/EasyNetQ.Management.Client/blob/uploadNuGet/appveyor.yml#L25-L31).  The server URI and ApiKey (secured with https://ci.appveyor.com/tools/encrypt) would have to be replaced with yours. :)

**Comment by micdenny** [comment]
we need to create a package with version similar to `2.0.0-netcore0001`

**Comment by conniey** [comment]
@micdenny Thanks for letting me know!  I was able to get this working with the `dotnet-gitversion` tool I saw in the project.json.  A few questions before I submit a PR:

* Should the `[AssemblyVersion()]` be updated or keep the one that gets compiled with [Version.cs](https://github.com/EasyNetQ/EasyNetQ.Management.Client/blob/master/Source/Version.cs)?
* .NET CLI supports creating a NuGet package, using `dotnet pack` and it magically figures out the nuspec based on your project.json, etc.  However, I noticed that [`Build\EasyNetQ.proj`](https://github.com/EasyNetQ/EasyNetQ.Management.Client/blob/master/Build/EasyNetQ.proj#L45-L49) merges the management client into a single dll.  So, I am wondering if that behaviour is something you want to keep?

**Comment by sysmatrix1** [comment]
micdenny and team - Thanks so much for your continued efforts to support EasyNetQ. Can you please publish  a .NET core compatible nuget package of EasyNetQ.Management.Client to myget ?

**Comment by zidad** [comment]
@sysmatrix1 Are you encountering problems with the one that is currently on myget already? But I agree it should be released to nuget soon.

https://www.myget.org/feed/easynetq-unstable/package/nuget/EasyNetQ.Management.Client

**Comment by wonea** [comment]
Would very much appreciate a NETSTANDARD build, especially as the main library has just gone stable with NETSTANDARD support.

**Comment by zidad** [comment]
@wonea I would very much like that too, but as it requires conversion to the new VS2017 format, it's too much work for me now. Feel free to submit a PR though :)
